### PR TITLE
Authentication and NLPSearchService TestClass

### PR DESCRIPTION
**Pull request by tanvi029**

### CODE REVIEW

**Comment by kailash** [comment]
ok to test

**Comment by kailash** [comment]
retest this please

**Comment by tanvi029** [comment]
[catalog-server-coverage.pdf](https://github.com/datakaveri/iudx-catalogue-server/files/9379598/catalog-server-coverage.pdf)

**Comment by kailash** [comment]
retest this please

File: src/test/java/iudx/catalogue/server/apiserver/util/ExceptionHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    @DisplayName("Test handle method with RuntimeException")
+    public void testHandle(VertxTestContext vertxTestContext) {
+        when(routingContext.failure()).thenReturn(failure);
+
+        when(routingContext.response()).thenReturn(httpServerResponse);
+        when(httpServerResponse.setStatusCode(400)).thenReturn(httpServerResponse);
+        when(httpServerResponse.putHeader(anyString(), anyString())).thenReturn(httpServerResponse);
+        ExceptionHandler exceptionHandler = new ExceptionHandler();
+        exceptionHandler.handle(routingContext);
+
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/ExceptionHandlerTest.java:43]: use assertions, `assertThrows()` in this case

File: src/test/java/iudx/catalogue/server/apiserver/util/ExceptionHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        when(routingContext.failure()).thenReturn(failure);
+        when(failure.getLocalizedMessage()).thenReturn("dummy msg");
+        when(routingContext.request()).thenReturn(httpServerRequest);
+        when(httpServerRequest.uri()).thenReturn(ROUTE_ITEM);
+        when(routingContext.response()).thenReturn(httpServerResponse);
+        when(httpServerResponse.setStatusCode(500)).thenReturn(httpServerResponse);
+        when(httpServerResponse.putHeader(anyString(),anyString())).thenReturn(httpServerResponse);
+        when(httpServerResponse.end(anyString())).thenReturn(voidFuture);
+        ExceptionHandler exceptionHandler=new ExceptionHandler();
+        exceptionHandler.handleDecodeException(routingContext);
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/ExceptionHandlerTest.java:59]: use assertions

File: src/test/java/iudx/catalogue/server/apiserver/util/ExceptionHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        when(routingContext.failure()).thenReturn(failure);
+        when(failure.getLocalizedMessage()).thenReturn("dummy msg");
+        when(routingContext.request()).thenReturn(httpServerRequest);
+        when(httpServerRequest.uri()).thenReturn(ROUTE_SEARCH);
+        when(routingContext.response()).thenReturn(httpServerResponse);
+        when(httpServerResponse.setStatusCode(500)).thenReturn(httpServerResponse);
+        when(httpServerResponse.putHeader(anyString(),anyString())).thenReturn(httpServerResponse);
+        when(httpServerResponse.end(anyString())).thenReturn(voidFuture);
+        ExceptionHandler exceptionHandler=new ExceptionHandler();
+        exceptionHandler.handleDecodeException(routingContext);
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/ExceptionHandlerTest.java:76]: use assertions

File: src/test/java/iudx/catalogue/server/apiserver/util/ExceptionHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/ExceptionHandlerTest.java:93]: use assertions

File: src/test/java/iudx/catalogue/server/apiserver/util/ExceptionHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    @DisplayName("Test handleClassCastException method")
+    public void testHandleClassCastException(VertxTestContext vertxTestContext) {
+        when(routingContext.failure()).thenReturn(failure);
+        when(failure.getLocalizedMessage()).thenReturn("dummy msg");
+        when(routingContext.response()).thenReturn(httpServerResponse);
+        when(httpServerResponse.setStatusCode(400)).thenReturn(httpServerResponse);
+        when(httpServerResponse.putHeader(anyString(),anyString())).thenReturn(httpServerResponse);
+        when(httpServerResponse.end(anyString())).thenReturn(voidFuture);
+        ExceptionHandler exceptionHandler=new ExceptionHandler();
+        exceptionHandler.handleClassCastException(routingContext);
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/ExceptionHandlerTest.java:107]: use assertions

File: src/test/java/iudx/catalogue/server/apiserver/util/QueryMapperTest.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        assertEquals(new JsonObject().put(COORDINATES,value),queryMapper.map2Json(queryParameters));
+        vertxTestContext.completeNow();
+    }
+    @Test
+    @DisplayName("Test map2Json method with  Q_Value key")
+    public void testMap2JsonQValue(VertxTestContext vertxTestContext) {
+        MultiMap queryParameters=MultiMap.caseInsensitiveMultiMap();
+        queryParameters.add(Q_VALUE,"!");
+        QueryMapper queryMapper=new QueryMapper();
+        queryMapper.map2Json(queryParameters);
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/QueryMapperTest.java:66]: use assertions

File: src/test/java/iudx/catalogue/server/apiserver/util/QueryMapperTest.java [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        vertxTestContext.completeNow();
+    }
+    @Test
+    @DisplayName("Test map2Json method with [] key")
+    public void testMap2JsonWithBracket(VertxTestContext vertxTestContext) {
+        MultiMap queryParameters=MultiMap.caseInsensitiveMultiMap();
+        queryParameters.add("dummy key","[abcd]");
+        QueryMapper queryMapper=new QueryMapper();
+        queryMapper.map2Json(queryParameters);
+        vertxTestContext.completeNow();
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/QueryMapperTest.java:76]: use assertions

File: src/test/java/iudx/catalogue/server/apiserver/util/QueryMapperTest.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    }
+    @Test
+    @DisplayName("Test map2Json method with exceptAttribute value")
+    public void testMap2JsonExceptAttributeValue(VertxTestContext vertxTestContext) {
+        MultiMap queryParameters=MultiMap.caseInsensitiveMultiMap();
+        queryParameters.add(COORDINATES,"[hhv]");
+        QueryMapper queryMapper=new QueryMapper();
+        queryMapper.map2Json(queryParameters);
+        vertxTestContext.completeNow();
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/QueryMapperTest.java:85]: use assertions

File: src/test/java/iudx/catalogue/server/apiserver/util/QueryMapperTest.java [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+    }
+    @Test
+    @DisplayName("Test map2Json method with  Q_Value key and q value")
+    public void testMap2JsonQ_Value(VertxTestContext vertxTestContext) {
+        MultiMap queryParameters=MultiMap.caseInsensitiveMultiMap();
+        queryParameters.add(Q_VALUE,"q");
+        QueryMapper queryMapper=new QueryMapper();
+        queryMapper.map2Json(queryParameters);
+        vertxTestContext.completeNow();
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/QueryMapperTest.java:94]: use assertions

File: src/test/java/iudx/catalogue/server/apiserver/util/QueryMapperTest.java [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+    @DisplayName("Test handle method with Q_VALUE GEOMETRY PROPERTY FILTER")
+    public void testMap2JsonAttributes(VertxTestContext vertxTestContext) {
+        MultiMap queryParameters=MultiMap.caseInsensitiveMultiMap();
+        queryParameters.add(Q_VALUE,"q");
+        queryParameters.add(GEOMETRY,"q");
+        queryParameters.add(PROPERTY,"q");
+        queryParameters.add(FILTER,"q");
+        QueryMapper queryMapper=new QueryMapper();
+        queryMapper.map2Json(queryParameters);
+        vertxTestContext.completeNow();
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/QueryMapperTest.java:106]: use assertions

File: src/test/java/iudx/catalogue/server/apiserver/util/RespBuilderTest.java [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+@ExtendWith(MockitoExtension.class)
+@ExtendWith(VertxExtension.class)
+public class RespBuilderTest {
+    @Test
+    @DisplayName("Test withDetail method")
+    public void testWithDetail(VertxTestContext vertxTestContext) {
+        RespBuilder respBuilder = new RespBuilder();
+        String detail = "dummy";
+        respBuilder.withDetail(detail);
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/RespBuilderTest.java:18]: use assertions

File: src/test/java/iudx/catalogue/server/apiserver/util/RespBuilderTest.java [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @Test
+    @DisplayName("Test withResult method")
+    public void testWithResult(VertxTestContext vertxTestContext) {
+        RespBuilder respBuilder = new RespBuilder();
+        String id="dummy id";
+        String method="dummy method";
+        String status="dummy status";
+        respBuilder.withResult(id,method,status);
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/RespBuilderTest.java:30]: use assertions

File: src/test/java/iudx/catalogue/server/apiserver/util/RespBuilderTest.java [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+    }
+    @Test
+    @DisplayName("Test withResult method with detail parameter")
+    public void testWithResultDetail(VertxTestContext vertxTestContext) {
+        RespBuilder respBuilder = new RespBuilder();
+        String id="dummy id";
+        String method="dummy method";
+        String status="dummy status";
+        String detail="dummy detail";
+        respBuilder.withResult(id,method,status,detail);
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/RespBuilderTest.java:42]: use assertions

File: src/test/java/iudx/catalogue/server/apiserver/util/RespBuilderTest.java [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+        vertxTestContext.completeNow();
+
+    }
+    @Test
+    @DisplayName("Test getJsonResponse method")
+    public void testGetJsonResponse(VertxTestContext vertxTestContext) {
+        RespBuilder respBuilder = new RespBuilder();
+
+        respBuilder.getJsonResponse();
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/apiserver/util/RespBuilderTest.java:51]: use assertions

File: src/test/java/iudx/catalogue/server/auditing/util/QueryBuilderTest.java [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+    @Mock
+    JsonObject request;
+
+    @Test
+    @Description("test buildReadQuery when request does not contain user id ")
+    public void testBuildReadQueryNoUSER_ID(VertxTestContext vertxTestContext){
+        request=new JsonObject();
+        queryBuilder=new QueryBuilder();
+        queryBuilder.buildReadQuery(request);
+        vertxTestContext.completeNow();
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/auditing/util/QueryBuilderTest.java:28]: use assertions

File: src/test/java/iudx/catalogue/server/auditing/util/QueryBuilderTest.java [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+    @Test
+    @Description("test buildReadQuery when request contains METHOD and ENDPOINT key")
+    public void testBuildReadQuery(VertxTestContext vertxTestContext) {
+        queryBuilder=new QueryBuilder();
+        request=new JsonObject();
+        request.put(METHOD,"httpMethod");
+        request.put(ENDPOINT,"endPoint");
+        request.put(USER_ID,"userID");
+        request.put(API,"api");
+        queryBuilder.buildReadQuery(request);
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/auditing/util/QueryBuilderTest.java:39]: use assertions

File: src/test/java/iudx/catalogue/server/authenticator/AuthenticationServiceImplTest.java [hunk 17]
```diff
@@ -0,0 +1,174 @@
...
+            @Override
+            public AsyncResult<HttpResponse<Buffer>> answer(InvocationOnMock arg0) throws Throwable {
+
+                ((Handler<AsyncResult<HttpResponse<Buffer>>>)arg0.getArgument(1)).handle(httpResponseAsyncResult);
+                return null;
+            }
+        }).when(httpRequest).sendJsonObject(any(),any());
+
+        authenticationService=new AuthenticationServiceImpl(webClient,authHost);
+        authenticationService.tokenInterospect(request,authenticationInfo,handler);
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/authenticator/AuthenticationServiceImplTest.java:74]: use assertions and use `verify()` to test exact interactions with mocks.

File: src/test/java/iudx/catalogue/server/authenticator/AuthenticationServiceImplTest.java [hunk 18]
```diff
@@ -0,0 +1,174 @@
...
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/authenticator/AuthenticationServiceImplTest.java:115]: use assertions and use verify() to test exact interactions with mocks.

File: src/test/java/iudx/catalogue/server/authenticator/AuthenticationServiceImplTest.java [hunk 19]
```diff
@@ -0,0 +1,174 @@
...
+    @Test
+    @Description("testing the method validateAuthInfo")
+    public void testIsPermittedMethodTrue(VertxTestContext vertxTestContext) {
+
+        String authHost="dummy";
+        authenticationService=new AuthenticationServiceImpl(webClient,authHost);
+        JsonArray methods=new JsonArray();
+        methods.add("*");
+        String operation="dummy";
+        authenticationService.isPermittedMethod(methods,operation);
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/authenticator/AuthenticationServiceImplTest.java:128]: use assertions

File: src/test/java/iudx/catalogue/server/authenticator/AuthenticationServiceImplTest.java [hunk 20]
```diff
@@ -0,0 +1,174 @@
...
+    @Test
+    @Description("testing the method validateAuthInfo")
+    public void testIsPermittedMethodFalse(VertxTestContext vertxTestContext) {
+
+        String authHost="dummy value";
+        authenticationService=new AuthenticationServiceImpl(webClient,authHost);
+        JsonArray methods=new JsonArray();
+        methods.add("dummyy");
+        String operation="dummy";
+        authenticationService.isPermittedMethod(methods,operation);
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/authenticator/AuthenticationServiceImplTest.java:140]: use assertions

File: src/test/java/iudx/catalogue/server/authenticator/AuthenticationServiceImplTest.java [hunk 21]
```diff
@@ -0,0 +1,174 @@
...
+    public void testValidateAuthInfoBlank(VertxTestContext vertxTestContext){
+        String authHost="dummy";
+        authenticationInfo=new JsonObject();
+        request=new JsonObject();
+        authenticationInfo.put(TOKEN,"");
+        authenticationInfo.put(OPERATION,"");
+        request.put(PROVIDER,"dummy");
+        AuthenticationServiceImpl.webClient=mock(WebClient.class);
+        authenticationService=new AuthenticationServiceImpl(webClient,authHost);
+        authenticationService.tokenInterospect(request,authenticationInfo,handler);
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/authenticator/AuthenticationServiceImplTest.java:154]: use assertions and use verify() to test exact interactions with mocks.

File: src/test/java/iudx/catalogue/server/authenticator/AuthenticationServiceImplTest.java [hunk 22]
```diff
@@ -0,0 +1,174 @@
...
+    public void testValidateAuthInfoRegex(VertxTestContext vertxTestContext){
+        String authHost="dummy";
+        authenticationInfo=new JsonObject();
+        request=new JsonObject();
+        authenticationInfo.put(TOKEN,"!!");
+        authenticationInfo.put(OPERATION,"dummy");
+        request.put(PROVIDER,"dummy");
+        AuthenticationServiceImpl.webClient=mock(WebClient.class);
+        authenticationService=new AuthenticationServiceImpl(webClient,authHost);
+        authenticationService.tokenInterospect(request,authenticationInfo,handler);
```
Comment (kailash) [inline src/test/java/iudx/catalogue/server/authenticator/AuthenticationServiceImplTest.java:169]: use assertions

**Review by kailash — CHANGES_REQUESTED** [review]
- Use assertions(`assertEquals` & `assertThrows`) to check the return values in test cases.
- In case of mocks use `verify()` to verify the interactions with mocks.

**Pull request merged**